dev/sbt-plugin/src/sbt-test/sbt-plugin/projections-happy-path/hello/src/main/java/com/example/hello/impl/HelloServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.example.hello.impl;
+
+import akka.Done;
+import akka.NotUsed;
+import com.example.hello.api.HelloService;
+import com.example.hello.impl.HelloCommand.Hello;
+import com.example.hello.impl.HelloCommand.UseGreetingMessage;
+import com.example.hello.impl.readsides.StartedProcessor;
+import com.example.hello.impl.readsides.StoppedProcessor;
+import com.lightbend.lagom.javadsl.api.ServiceCall;
+import com.lightbend.lagom.javadsl.persistence.PersistentEntityRef;
+import com.lightbend.lagom.javadsl.persistence.PersistentEntityRegistry;
+import com.lightbend.lagom.javadsl.persistence.ReadSide;
+import com.lightbend.lagom.javadsl.projection.Projections;
+import com.lightbend.lagom.projection.State;
+
+import javax.inject.Inject;
+
+public class HelloServiceImpl implements HelloService {
+    private final PersistentEntityRegistry persistentEntityRegistry;
+    private final Projections projections;
+    private final StartedProcessor startedProcessor;
+    private final StoppedProcessor stoppedProcessor;
+
+    @Inject
+    public HelloServiceImpl(
+        PersistentEntityRegistry persistentEntityRegistry,
+        Projections projections,
+        ReadSide readSide,
+
+        StartedProcessor startedProcessor,
+        StoppedProcessor stoppedProcessor
+    ) {
+        this.projections = projections;
+        this.startedProcessor = startedProcessor;
+        this.stoppedProcessor = stoppedProcessor;
+
+        // This sample is setup to not start the projections eagerly (see application.conf) but we do
+        // start one of the projections programmatically.
+        projections.startAllWorkers(StartedProcessor.NAME);
+
+        readSide.register(StartedProcessor.class);
+        readSide.register(StoppedProcessor.class);
```

Comment:
The three lines above are the key step on the test: request the workers of a projection to be started _before_ registering the processor. :-)

Reply:
Can we have a code comment then?